### ISSUE TITLE
Markdown Documentation Generation

### ISSUE THREAD

**Issue by latenitefilms** [Hammerspoon/hammerspoon#1872]
@cmsj - I need your help if you have a sec!

I've modified one of the markdown templates as follows:

```
# [docs](index.md) » {{ module.name }}
---

{{ module.doc }}
{% if module["submodules"]|length > 0 %}

## Submodules
{% for submodule in module["submodules"] %}
 * [{{ module.name }}.{{ submodule }}]({{ module.name }}.{{ submodule }}.md)
{% endfor %}
{% endif %}

## API Overview
{% for type in type_order %}
{# Considering: {{ type }} ({{ module[type]|length }}) #}
{% if module[type]|length > 0 %}
* {{ type }}s - {{ type_desc[type] }}
{% for item in module[type] %}
 * [{{ item.name }}](#{% filter lower %}{{ item.name }}{% endfilter %})
{% endfor %}
{% endif %}
{% endfor %}

## API Documentation

{% for type in type_order %}{% if module[type]|length > 0 %}
### {{ type}}s

{% for item in module[type] %}
#### [{{ item.name }}](#{% filter lower %}{{ item.name }}{% endfilter %})
| <span style="float: left;">**Signature**</span> | <span style="float: left;">`{{ item.def }}` </span>                                                          |
| -----------------------------------------------------|---------------------------------------------------------------------------------------------------------|
| **Type**                                             | {{ item.type }}                                                                                         |
| **Description**                                      | {{ item.desc }}                                                                                         |
{% if "parameters" in item %}
| **Parameters**                                       | <ul>{% for parameter in item.parameters %}<li>{{ parameter | replace(" * ","") }}</li>{% endfor %}</ul> |
{% endif %}
{% if "returns" in item %}
| **Returns**                                          | <ul>{% for return in item.returns %}<li>{{ return | replace(" * ","") }}</li>{% endfor %}</ul>          |
{% endif %}
{% if "notes" in item %}
| **Notes**                                            | <ul>{% for note in item.notes %}<li>{{ note | replace(" * ","") }}</li>{% endfor %}</ul>                |
{% endif %}

{% endfor %}{% endif %}{% endfor %}
```

I'd now like to enable markdown within HTML tags.

I believe `mistune` can do this via:

```
markdown = mistune.Markdown(parse_block_html=True,parse_inline_html=True)
```

However, when I try and add these tags in the `process_markdown` function in `build_docs.py` it doesn't help. I've tried a few other things, but I can't quite work it out.

Any tips or tricks?

Thanks heaps mate! Hope all is well on the other side of the world!

**Comment by cmsj** [comment]
I'm not quite sure I understand what it is you're trying to achieve

**Comment by latenitefilms** [comment]
@cmsj - I'm using `build_docs.py` to generate markdown documentation that's stored on GitBook.

For example, when I use the following Terminal command:

```
build_docs.py -o build/CommandPost-Docs/cp/ --standalone --debug --markdown ../CommandPost/src/extensions/cp/
```

...it will convert this inline code:

```lua
--- cp.prop.AND(...) -> cp.prop
--- Function
--- Returns a new `cp.prop` which will be `true` if all `cp.prop` instances passed into the function return a `truthy` value.
---
--- Parameters:
---  * `...`        - The list of `cp.prop` instances to 'AND' together.
---
--- Returns:
---  * a `cp.prop` instance.
---
--- Notes:
---  * The value of this instance will resolve by lazily checking the `value` of the contained `cp.prop` instances in the order provided. The first `falsy` value will be returned. Otherwise the last `truthy` value is returned.
---  * The instance is **immutable**.
---  * Once you have created an 'AND', you cannot 'OR' as a method. Eg, this will fail: `prop.TRUE():AND(prop:FALSE()):OR(prop.TRUE())`. This is to avoid ambiguity as to whether the 'AND' or 'OR' takes precedence. Is it `(true and false) or true` or `true and (false or true)`?.
---  * To combine 'AND' and 'OR' values, group them together when combining. Eg:
---  ** `(true and false) or true`: `prop.OR( prop.TRUE():AND(prop.FALSE()), prop.TRUE() )`
---  ** `true and (false or true)`: `prop.TRUE():AND( prop.FALSE():OR(prop.TRUE()) )`
```

...to this:

<img width="1325" alt="screen shot 2018-06-21 at 5 56 46 am" src="https://user-images.githubusercontent.com/22286696/41681554-4db12a46-7518-11e8-9bce-ec0b9c04e91e.png">

Notice that the notes lines have some markdown that's not being processed as HTML. This is what I would like to change.

The reason it doesn't process this content as markdown is because by default `mistune` doesn't process markdown if it's within a HTML tag. In the template I'm using:

```
{% for item in module[type] %}
#### [{{ item.name }}](#{% filter lower %}{{ item.name }}{% endfilter %})
| <span style="float: left;">**Signature**</span> | <span style="float: left;">`{{ item.def }}` </span>                                                          |
| -----------------------------------------------------|---------------------------------------------------------------------------------------------------------|
| **Type**                                             | {{ item.type }}                                                                                         |
| **Description**                                      | {{ item.desc }}                                                                                         |
{% if "parameters" in item %}
| **Parameters**                                       | <ul>{% for parameter in item.parameters %}<li>{{ parameter | replace(" * ","") }}</li>{% endfor %}</ul> |
{% endif %}
{% if "returns" in item %}
| **Returns**                                          | <ul>{% for return in item.returns %}<li>{{ return | replace(" * ","") }}</li>{% endfor %}</ul>          |
{% endif %}
{% if "notes" in item %}
| **Notes**                                            | <ul>{% for note in item.notes %}<li>{{ note | replace(" * ","") }}</li>{% endfor %}</ul>                |
{% endif %}
```

So basically, I want to work out a way to process markdown as markdown even if it's in a `<ul>...</ul>` HTML block.

Does that make sense?

**Comment by latenitefilms** [comment]
To clarify, the problem is, markdown doesn't support lists within tables, so we're using HTML tags to get around that issue, but GitBook doesn't process markdown within HTML tags - so confusing!

**Comment by latenitefilms** [comment]
I THINK I've solve this by importing `markdown` into `jinja2`.
